Sources/IMAPlugin.swift

```diff
@@ -421,14 +429,17 @@ enum IMAState: Int, StateProtocol, CustomStringConvertible {
                 self.discardAdBreak(adsManager: adsManager)
             } else {
                 var adEvent = AdEvent.AdLoaded()
-                if event.ad != nil {
-                    let adInfo = PKAdInfo(ad: event.ad,
-                                          podCount: adsManager?.adCuePoints.count,
+                
+                if let ad = event.ad {
```

Comment:
exctract to method

Reply:
Same problem like in `IMADAIPlugin.swift`